url7f43a4752f_shalynnho_pokemonfactory_tgz-pJ8-Networking_ServerJ8/src/agent/StandAgent.java

```diff
@@ -0,0 +1,389 @@
+
+	private final ArrayList<MyKit> MyKits;
+
+	private int numKitsToMake;
+	private int numKitsMade;
+	private int oldLocForKitBeingInspected;
+	private boolean start;
+	private StandStatus status;
+
+	private enum StandStatus {
+		IDLE, NEED_TO_INITIALIZE, KIT_REQUESTED, DONE
+	};
+
+	/**
+	 * Inner class encapsulates kit and adds states relevant to the stand
+	 * 
+	 * @author dpaje
+	 */
+	private class MyKit {
+		public Kit kit;
+		public KitStatus KS;
+
+		public MyKit(Kit k) {
+			this.kit = k;
+			this.KS = KitStatus.RECEIVED;
+		}
+	}
+
+	public enum KitStatus {
+		HOLDING, RECEIVED, PLACED_ON_STAND, ASSEMBLED, MARKED_FOR_INSPECTION, AWAITING_INSPECTION, INSPECTED, SHIPPED, DELIVERED;
+	};
+
+	/**
+	 * Constructor for StandAgent class
+	 * 
+	 * @param name
+	 *            name of the stand
+	 */
+	public StandAgent(String name) {
+		super();
+
+		this.name = name;
+		numKitsToMake = 0;
+		numKitsMade = 0;
+		start = false;
+		status = StandStatus.IDLE;
+
+		MyKits = new ArrayList<MyKit>();
+		MyKits.add(0, null);
+		MyKits.add(1, null);
+		MyKits.add(2, null);
+	}
+
+	/*
+	 * Messages
+	 */
+
+	@Override
+	public void msgMakeKits(int numKits) {
+		print("Received msgMakeKits");
+		numKitsToMake = numKits;
+		numKitsMade = 0;
+		status = StandStatus.NEED_TO_INITIALIZE;
+		start = true;
+		stateChanged();
+	}
+
+	@Override
+	public void msgHereIsKit(Kit k, int destination) {
+		print(k.toString());
+		print("Received msgHereIsKit with destination " + destination);
+		status = StandStatus.IDLE;
+		if (MyKits.get(0) != null) {
+			if (MyKits.get(0).kit == k) {
+				print(k.toString() + " must have failed inspection");
+				MyKits.set(0, null);
+			}
+		}
+		MyKits.set(destination, new MyKit(k));
+		stateChanged();
+	}
+
+	@Override
+	public void msgKitAssembled(Kit k) {
+		print("Received msgKitAssembled");
+		for (MyKit mk : MyKits) {
+			if (mk != null) {
+				if (mk.KS != KitStatus.HOLDING) {
+					if (mk.kit == k) {
+						print(k.toString() + " was assembled.");
+						mk.KS = KitStatus.ASSEMBLED;
+						break;
+					}
+				}
+			}
+		}
+		stateChanged();
+	}
+
+	@Override
+	public void msgMovedToInspectionArea(Kit k, int oldLocation) {
+		print("Received msgMovedToInspectionArea");
+		if (MyKits.get(oldLocation) == null) {
+			print("OMGNOOOO");
+		}
+		MyKits.set(0, MyKits.get(oldLocation));
+		print(MyKits.get(0).kit.toString() + " is now in the inspection area");
+
+		oldLocForKitBeingInspected = oldLocation;
+		// MyKits.set(oldLocation, null);
+
+		stateChanged();
+	}
+
+	@Override
+	public void msgShippedKit() {
+		print("Received msgShippedKit");
+		numKitsMade++;
+		MyKits.set(0, null);
+		// MyKits.get(0).KS = KitStatus.SHIPPED;
+		MyKits.set(oldLocForKitBeingInspected, null);
+		print(numKitsToMake - numKitsMade + " kits left to make");
+		fcs.msgShippedKit();
+		stateChanged();
+	}
+
+	/*
+	 * Scheduler
+	 * 
+	 * @see agent.Agent#pickAndExecuteAnAction()
+	 */
+	@Override
+	public boolean pickAndExecuteAnAction() {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 26. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.